### ISSUE TITLE
Deprecated: str_replace und explode bei PHP 8.1

### ISSUE THREAD

**Issue by mbodon** [FriendsOfREDAXO/mblock#160]
Hallo, bekomme seit nem Update auf 8.1 und neustes mForm & mBlock eine Fehlermeldung wenn ich ein Modul mit Tabs editiere: `Deprecated: str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in redaxo/src/addons/mform/lib/MForm/MFormElements.php on line 65 ` … gibt’s da eine Lösung?

scheint mir zu helfen wenn ich die Zeile 64 austausche durch : `if(!is_int($id) && $id !== null) {`

aber dann meckert explode : `Deprecated: explode(): Passing null to parameter #2 ($string) of type string is deprecated in redaxo/src/addons/mform/lib/MForm/Handler/MFormElementHandler.php on line 28` da hat mir folgender zusätzlich Code direkt nach Zeile 25 geholfen  … `if ($varId === null) { $varId = ""; }` (hoffentlich)

**Comment by alxndr-w** [comment]
> gibt’s da eine Lösung?

Ja, einfach einen PR einreichen mit der Lösung. Beschreib' das Problem notfalls Github Copilot.

Du musst auf String prüfen und/oder herausfinden, wie es überhaupt zu `null` als Übergabe kommen kann.

**Comment by tyrant88** [comment]
ist ja ne mform Sache gewesen, also mach ich hier mal zu.
